### PR TITLE
[mtoh] Split pre and post scene passes so camera guides go above Hydra.

### PR DESCRIPTION
**Pull request by marsupial**
#525

### CODE REVIEW

File: lib/mayaUsd/render/mayaToHydra/renderOverrideUtils.h [hunk 1]
```diff
@@ -66,13 +66,17 @@ class HdMayaManipulatorRender : public MHWRender::MSceneRender {
...
+        return MHWRender::MSceneRender::getObjectTypeExclusions() | ~(kExcludeManipulators | kExcludeHoldOuts);
     }
 
     MHWRender::MClearOperation& clearOperation() override {
         mClearOperation.setMask(MHWRender::MClearOperation::kClearNone);
         return mClearOperation;
     }
+
+    MSceneFilterOption renderFilterOverride() override {
+        return MSceneFilterOption(MHWRender::MSceneRender::renderFilterOverride() | kRenderPostSceneUIItems);
```
Comment (HdC-adsk) [inline lib/mayaUsd/render/mayaToHydra/renderOverrideUtils.h:78]: Can you return kRenderPostSceneUIItems directly? From API doc, MSceneRender::renderFilterOverride() returns kNoSceneFilterOverride which equals 0.

File: lib/mayaUsd/render/mayaToHydra/renderOverrideUtils.h [hunk 2]
```diff
@@ -66,13 +66,17 @@ class HdMayaManipulatorRender : public MHWRender::MSceneRender {
         : MHWRender::MSceneRender(name) {}
 
     MUint64 getObjectTypeExclusions() override {
-        return ~MHWRender::MFrameContext::kExcludeManipulators;
+        return MHWRender::MSceneRender::getObjectTypeExclusions() | ~(kExcludeManipulators | kExcludeHoldOuts);
```
Comment (HdC-adsk) [inline lib/mayaUsd/render/mayaToHydra/renderOverrideUtils.h:69]: Is this change relevant to the issue? Why do you exclude holdouts?

`MHWRender::MSceneRender::getObjectTypeExclusions()` returns 0, so there is no need to use it in the bitmask op.

**Comment by HdC-adsk** [comment]
I don't know well enough about mtoh implementation to review the impact to any existing workflow, adding @elrond79 to review too.

**Comment by marsupial** [comment]
I just called through to the base classes as I didn't know they returned 0.  So I can test to see if removing them still works.

The other change is relevant, as kExcludeHoldOuts is what winds up drawing the camera guides from what I've seen.

**Comment by elrond79** [comment]
Haven't tested the code, but the code changes make sense to me, assuming that kExcludeHoldOuts is where camera guides are grouped.  (Which seems odd, but I'll take your word for it!)

**Comment by HdC-adsk** [comment]
Yes, a bit odd. Maybe adding a comment to explain the purpose of excluding holdout would be better.

**Comment by HdC-adsk** [comment]
> 
> 
> I just called through to the base classes as I didn't know they returned 0. So I can test to see if removing them still works.
> 
> The other change is relevant, as kExcludeHoldOuts is what winds up drawing the camera guides from what I've seen.

@marsupial How is the test doing?

**Comment by marsupial** [comment]
Sadly no time! Monday for sure.

**Comment by kxl-adsk** [comment]
@marsupial I see you (force) pushed changes two days ago to the branch. We are not receiving notification for this action from GitHub. For future PRs, please leave a comment in the PR when reviewers should review new changes...or you can re-request the review from a specific person which will send a notification.
